### ISSUE TITLE
Dynamically generated models should inherit the Config of their __base__

### ISSUE THREAD

**Issue by aalexandrov** [samuelcolvin/pydantic#283]
# Bug

Thanks for the great library, @samuelcolvin! I have been using it quite extensively in the past weeks and hope to contribute a possible bug (+ a suggested fix).

* OS: **Mac OS**
* Python version `import sys; print(sys.version)`: **3.6.4**
* Pydantic version `import pydantic; print(pydantic.VERSION)`: **0.12.1**

A self contained code snippet describing the bug:

```py
import pydantic

class Foo:
    pass

# OK
class Bar(pydantic.BaseModel):
    x: Foo
    class Config(pydantic.BaseConfig):
        arbitrary_types_allowed = True

# throws RuntimeError: no validator found for <class '__main__.Foo'>
Baz = pydantic.create_model('Bar', __base__=Bar, y=(Foo, ...))
```

I think that [this line](https://github.com/samuelcolvin/pydantic/blob/master/pydantic/main.py#L402) should be changed either to

```python
config = __config__ or __base__.Config
```

or to

```python
config = __config__ or __base__.Config or BaseConfig
```

if the meta-class does not set `__base__.Config = BaseConfig` by default.

**Comment by aalexandrov** [comment]
Sorry, I wanted to link [line 402](https://github.com/samuelcolvin/pydantic/blob/master/pydantic/main.py#L402).

**Comment by samuelcolvin** [comment]
fixed in **v0.16.1**, #320
